src/test/java/com/cegeka/academy/repository/EventRepositoryTest.java

```diff
@@ -38,6 +40,50 @@ public void testAddEvent() {
         assertThat(eventTest.getPublic()).isEqualTo(true);
         assertThat(eventTest.getName()).isEqualTo(event.getName());
     }
+    @Test
+    public void testFindAllByCategoryId() {
+        User user = TestsRepositoryUtil.createUser("login", "anaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaana");
+        userRepository.save(user);
+        Address address = TestsRepositoryUtil.createAddress("Romania", "Bucuresti", "Splai", "333", "Casa", "Casa magica");
+        addressRepository.saveAndFlush(address);
+        Event event = TestsRepositoryUtil.createEvent("Ana are mere!", "KFC Krushers Party", true, address, user);
+        eventRepository.save(event);
+        Category category1=TestsRepositoryUtil.createCategory("Ana","description");
+        categoryRepository.save(category1);
+        event.getCategories().add(category1);
+        eventRepository.save(event);
+        Category category2=TestsRepositoryUtil.createCategory("Dana","description");
+        categoryRepository.save(category2);
+        event.getCategories().add(category2);
+        eventRepository.save(event);
+        List<Event>events=eventRepository.findAllByCategories_id(category1.getId());
+        assertThat(events.size()).isEqualTo(1);
+    }
+    @Test
+    public void testFindAllByEventId()
+    {
+        User user = TestsRepositoryUtil.createUser("login", "anaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaanaana");
+        userRepository.save(user);
+        Address address = TestsRepositoryUtil.createAddress("Romania", "Bucuresti", "Splai", "333", "Casa", "Casa magica");
+        addressRepository.saveAndFlush(address);
+        Event event = TestsRepositoryUtil.createEvent("Ana are mere!", "KFC Krushers Party", true, address, user);
+        eventRepository.save(event);
+        Category category1=TestsRepositoryUtil.createCategory("Ana","description");
+        categoryRepository.save(category1);
+        event.getCategories().add(category1);
+        eventRepository.save(event);
+        Category category2=TestsRepositoryUtil.createCategory("Dana","description");
+        categoryRepository.save(category2);
+        event.getCategories().add(category2);
+        eventRepository.save(event);
+        Category category3=TestsRepositoryUtil.createCategory("Doriana","description");
+        categoryRepository.save(category3);
+        event.getCategories().add(category3);
+        eventRepository.save(event);
+        List<Category>categories=categoryRepository.findAllByEvents_id(event.getId());
```

Comment:
Testarea metodelor din `CategoryRepository` ar trebui facuta in clasa de test specifica.